lib/spack/spack/cmd/style.py

```diff
@@ -507,7 +518,193 @@ def _bootstrap_dev_dependencies():
         spack.bootstrap.ensure_environment_dependencies()
 
 
+IS_PROBABLY_COMPILER = re.compile(r"%[a-zA-Z_][a-zA-Z0-9\-]")
+
+
+def _spec_str_reorder_compiler(idx: int, blocks: List[List[Token]]) -> None:
+    # only move the compiler to the back if it exists and is not already at the end
+    if not 0 <= idx < len(blocks) - 1:
+        return
+    # ensure there is at least one non-WS token after the compiler token
+    for block in blocks[idx + 1 :]:
+        if any(token.kind != SpecTokens.WS for token in block):
+            break
+    else:
+        return
```

Comment:
```suggestion
    if all(token.kind == SpecTokens.WS for block in blocks[idx + 1 :] for token in block):
        return
```
This seems a bit more straightforward to read, and I think it boils down to the same operations.